manifests/classic.pp

```diff
@@ -57,19 +57,16 @@
   $joinou                         = undef,
   Optional[String] $default_realm = undef,
   Array $additional_realms        = [],
-) inherits samba::params{
-
-
-  unless is_domain_name($realm){
+) inherits samba::params {
```

Comment:
my recommendation is to add the datatypes to the class parameters,that makes the whole file shorter and enabled puppet-strings to pick up the types for the automatic documentation.